gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/JobStatistics.java

```diff
@@ -0,0 +1,471 @@
+package com.google.gcloud.bigquery;
+
+import com.google.api.services.bigquery.model.JobStatistics2;
+import com.google.api.services.bigquery.model.JobStatistics3;
+import com.google.api.services.bigquery.model.JobStatistics4;
+import com.google.common.base.MoreObjects;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A Google BigQuery Job statistics.
+ */
+public class JobStatistics implements Serializable {
+
+  private static final long serialVersionUID = 1433024714741660399L;
+
+  private final Long creationTime;
+  private final Long endTime;
+  private final Long startTime;
+
+  /**
+   * A Google BigQuery Extract Job statistics.
+   */
+  public static class ExtractStatistics extends JobStatistics {
+
+    private static final long serialVersionUID = -1566598819212767373L;
+
+    private final List<Long> destinationUriFileCounts;
+
+    static final class Builder extends JobStatistics.Builder<ExtractStatistics, Builder> {
+
+      private List<Long> destinationUriFileCounts;
+
+      private Builder() {}
+
+      private Builder(com.google.api.services.bigquery.model.JobStatistics statisticsPb) {
+        super(statisticsPb);
+        this.destinationUriFileCounts = statisticsPb.getExtract().getDestinationUriFileCounts();
+      }
+
+      Builder destinationUriFileCounts(List<Long> destinationUriFileCounts) {
+        this.destinationUriFileCounts = destinationUriFileCounts;
+        return self();
+      }
+
+      @Override
+      ExtractStatistics build() {
+        return new ExtractStatistics(this);
+      }
+    }
+
+    private ExtractStatistics(Builder builder) {
+      super(builder);
+      this.destinationUriFileCounts = builder.destinationUriFileCounts;
+    }
+
+    /**
+     * Returns the number of files per destination URI or URI pattern specified in the extract job.
+     * These values will be in the same order as the URIs specified by
+     * {@link ExtractJobInfo#destinationUris()}.
+     */
+    public List<Long> destinationUriFileCounts() {
+      return destinationUriFileCounts;
+    }
+
+    @Override
+    protected MoreObjects.ToStringHelper toStringHelper() {
```

Comment:
You are right, `MoreObjects.` prefix is needed (not sure, why intellij marks it otherwise - any reason for not importing it?). Regardless, let's replace the protected with package scope here as well.